cpp/src/arrow/compute/api_scalar.h

```diff
@@ -43,7 +48,9 @@ namespace compute {
 /// \param[in] ctx the function execution context, optional
 /// \return the elementwise sum
 ARROW_EXPORT
-Result<Datum> Add(const Datum& left, const Datum& right, ExecContext* ctx = NULLPTR);
+Result<Datum> Add(const Datum& left, const Datum& right,
+                  ArithmeticOptions options = ArithmeticOptions(),
```

Comment:
We need to document this.
This breaks CI: https://github.com/apache/arrow/runs/786249626

Reply:
I fixed it in https://github.com/apache/arrow/pull/7492

Reply:
Thanks!